app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -50,9 +50,16 @@ def create_or_update_inventory!(inventory_collection, record_index, association)
       unique_index_keys = inventory_collection.manager_ref_to_cols
 
       association.find_each do |record|
-        # TODO(lsmola) the old code was able to deal with duplicate records, should we do that? The old data still can
-        # have duplicate methods, so we should clean them up. It will slow up the indexing though.
-        record_index[inventory_collection.object_index_with_keys(unique_index_keys, record)] = record
+        index = inventory_collection.object_index_with_keys(unique_index_keys, record)
```

Comment:
that means we will need a custom saving code for there cases, until we are able to generalize it

AFAIK only VmWare acts like this, so we sould be safe saving other Vms the standard way.